### ISSUE TITLE
Add ability to use the GPTL timers in scream C++ code

### ISSUE THREAD

**Issue by ndkeen** [E3SM-Project/scream#1524]
It looks like we aren't that far away from this working, but as Jeff and I dug a little deeper, it's clear we can improve the support.

Currently, for several key cpp files, I can `#include "gptl.h"` and add `GPTLstart("timer1");` (with a matching stop) and I will get the timer output in `run/HommeTime.0` mixed in with other timers (some from fortran) in expected way.

**Comment by jeff-cohere** [comment]
Arguably, this would be a good candidate for EKAT. What think you, @bartgol ?

**Comment by bartgol** [comment]
If EKAT added GPTL as a submodule, we run the risk of having several places where these files are located. Currently, it is available in CIME, scorpio, as well as Homme, with some logic here and there to avoid multiple compilation (I hope).

What we could do, is start by linking gptl to `scream_share` though. In standalone builds, we get it from building scorpio, while in CIME builds we're already wrapping the CIME-generated library in a cmake tgt.

**Comment by jeff-cohere** [comment]
I was just thinking it'd be nice to have one place where GPTL is built and made available. It's kind of a mess the way we use it now.

**Comment by bartgol** [comment]
I agree, there's too many entry point. But then, [this](https://menloservice.sandia.gov/https://xkcd.com/927/).

Also, I don't think EKAT should just provide modules. I think gptl can be provided by the `share` folder of e3sm.

**Comment by bartgol** [comment]
Fixed by #1567
